### PR TITLE
w3m: email UI

### PR DESCRIPTION
**Pull request by jakubuid**
UI for email login

### CODE REVIEW

File: product/web3modal/src/main/kotlin/com/walletconnect/web3/modal/ui/routes/connect/email/RegisterDevice.kt [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+            textAlign = TextAlign.Center,
+        )
+        Text(
+            text = "jakub@walletconnect.com", //TODO: Replace with email
+            style = Web3ModalTheme.typo.paragraph400,
+            textAlign = TextAlign.Center
+        )
+        VerticalSpacer(12.dp)
+        Text(
+            text = "The code expires in 20 minutes",
```
Comment (ignaciosantise) [inline product/web3modal/src/main/kotlin/com/walletconnect/web3/modal/ui/routes/connect/email/RegisterDevice.kt:52]:
```suggestion
            text = "The link expires in 20 minutes",
```

File: product/web3modal/src/main/kotlin/com/walletconnect/web3/modal/ui/routes/connect/email/RegisterDevice.kt [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+        horizontalAlignment = Alignment.CenterHorizontally
+    ) {
+        VerticalSpacer(24.dp)
+        Image(
+            painter = painterResource(id = R.drawable.ic_check_blue),
+            contentDescription = "Register Device"
+        )
+        VerticalSpacer(16.dp)
+        Text(
+            text = "Approve login the link we sent to",
```
Comment (ignaciosantise) [inline product/web3modal/src/main/kotlin/com/walletconnect/web3/modal/ui/routes/connect/email/RegisterDevice.kt:41]:
```suggestion
            text = "Approve the login link we sent to",
```

File: product/web3modal/src/main/kotlin/com/walletconnect/web3/modal/ui/routes/connect/email/ConfirmEmail.kt [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+            textAlign = TextAlign.Center,
+        )
+        Text(
+            text = "jakub@walletconnect.com", //TODO: Replace with email
+            style = Web3ModalTheme.typo.paragraph400,
+            textAlign = TextAlign.Center
+        )
+        VerticalSpacer(12.dp)
+        Text(
+            text = "The code expires in 10 minutes",
```
Comment (ignaciosantise) [inline product/web3modal/src/main/kotlin/com/walletconnect/web3/modal/ui/routes/connect/email/ConfirmEmail.kt:52]:
```suggestion
            text = "The code expires in 20 minutes",
```

File: product/web3modal/src/main/kotlin/com/walletconnect/web3/modal/ui/components/internal/root/Web3ModalRoot.kt [hunk 4]
```diff
@@ -56,7 +56,8 @@ internal fun Web3ModalRoot(
     val snackBarState = rememberSnackBarState(coroutineScope = scope)
     val emailInputState = rememberEmailInputState { input ->
         //todo: validate input, trigger magic to send email
-        rootState.navigateToRegisterDevice()
+//        rootState.navigateToRegisterDevice()
+        rootState.navigateToConfirmEmail()
```
Comment (ignaciosantise) [inline product/web3modal/src/main/kotlin/com/walletconnect/web3/modal/ui/components/internal/root/Web3ModalRoot.kt:60]: Here you should have an if condition to redirect to navigateToConfirmEmail or navigateToRegisterDevice, right?

**Comment by ignaciosantise** [comment]
@jakubuid can you update some screenshots of the views? or a video

File: product/web3modal/src/main/kotlin/com/walletconnect/web3/modal/ui/components/internal/root/Web3ModalRoot.kt (hunk 4 above)
Comment (jakubuid) [reply product/web3modal/src/main/kotlin/com/walletconnect/web3/modal/ui/components/internal/root/Web3ModalRoot.kt:60]: Yes - this PR is focused on mockups only, I'll add business logic for login in the next one

**Comment by jakubuid** [comment]
> @jakubuid can you update some screenshots of the views? or a video

Screenshot_20240404-080929
Screenshot_20240404-081200
Screenshot_20240404-081241

@ignaciosantise

File: product/web3modal/src/main/kotlin/com/walletconnect/web3/modal/ui/components/internal/email/webview/MagicWebViewWrapper.kt [hunk 5]
```diff
@@ -80,28 +79,25 @@ internal class EmailMagicWebViewWrapper(
 
         @JavascriptInterface
         fun postMessage(message: String) {
-
-            logger.log("kobe: postMessage $message")
+            logger.log("postMessage $message")
 
             if (message == "verify_ready") {
```
Comment (quetool) [inline product/web3modal/src/main/kotlin/com/walletconnect/web3/modal/ui/components/internal/email/webview/MagicWebViewWrapper.kt:84]: I don't think this is needed anymore, it's being a while I don't see this message coming. And in any case you have already your JSONObject(message) wrapped with try/catch block so maybe safe to delete?

**Review by quetool — APPROVED** [review]

**Comment by ignaciosantise** [comment]
One comment for the next PR:
According to Figma, the email input height is the same as the buttons below

Screenshot 2024-04-04 at 10 49 47 AM

**Review by ignaciosantise — APPROVED** [review]

**Comment by jakubuid** [comment]
> One comment for the next PR: According to Figma, the email input height is the same as the buttons below
> 
> Screenshot 2024-04-04 at 10 49 47 AM

@ignaciosantise did you measure it at some point? In code, both have 56.dp

**Comment by ignaciosantise** [comment]
In figma they are both also 56dp, but in the screenshot the email seems to have a different height

Screenshot 2024-04-04 at 12 13 58 PM

**Pull request merged**